meilisearch-http/src/routes/indexes/documents.rs

```diff
@@ -227,26 +227,51 @@ async fn document_addition(
 
     let (uuid, mut update_file) = index_scheduler.create_update_file()?;
 
-    // TODO: this can be slow, maybe we should spawn a thread? But the payload isn't Send+Sync :weary:
-    // push the entire stream into a `Vec`.
-    // If someone sends us a never ending stream we're going to block the thread.
-    // TODO: Maybe we should write it to a file to reduce the RAM consumption
-    // and then reread it to convert it to obkv?
-    let mut buffer = Vec::new();
+    let err: Result<SummarizedTaskView, MeilisearchHttpError> =
+        Err(MeilisearchHttpError::Payload(ReceivePayloadErr));
+
+    let temp_file = match NamedTempFile::new() {
+        Ok(temp_file) => temp_file,
+        Err(e) => {
+            error!("create a temporary file error: {}", e);
+            return err;
+        }
+    };
+    debug!("temp file path: {:?}", temp_file.as_ref());
+    let buffer_file = match temp_file.reopen() {
+        Ok(buffer_file) => buffer_file,
+        Err(e) => {
+            error!("reopen payload temporary file error: {}", e);
+            return err;
+        }
+    };
+    let mut buffer = BufWriter::new(buffer_file);
+    let mut buffer_write_size: usize = 0;
     while let Some(bytes) = body.next().await {
-        buffer.extend_from_slice(&bytes?);
+        match buffer.write(&bytes?) {
+            Ok(size) => buffer_write_size += size,
+            Err(e) => {
+                error!("bufWriter write error: {}", e);
+                return err;
+            }
+        }
     }
-    if buffer.is_empty() {
+
+    if let Err(e) = buffer.flush() {
+        error!("bufWriter flush error: {}", e);
+        return err;
+    };
```

Comment:
Would it be possible to always declare the whole `Err(MeilisearchHttpError::Payload(ReceivePayloadErr))` instead of declaring it as an `err` at the top, please?

Reply:
What do you think about storing the error information directly as a string in the `ReceivePayloadErr` variant?

Reply:
Storing a different error message to `ReceivePayloadErr` does not seem necessary here. Exceptions that cause storing paylod to a temporary file are almost always an OS or network problem for the user. What is more important to see in this case is the error log.

Reply:
Hey, sorry, but I'm 100% with @Kerollmops on this one.
> What is more important to see in this case is the error log.

Most users don't read or can't access their logs, they're going to report their bug from the HTTP error we return, and thus, we definitely want to be able to differentiate all the possible errors.

You should update the error type to something like that:
```
#[error("Error while writing the playload to disk: {}", .0)]
```

Also, we don't want to struggle with lifetimes in case of errors; store the message as a `String`.